x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/DateUtils.java

```diff
@@ -33,10 +54,21 @@ public static ZonedDateTime asDateTime(long millis) {
      * Parses the given string into a DateTime using UTC as a default timezone.
      */
     public static ZonedDateTime asDateTime(String dateFormat) {
-        return DateFormatters.from(UTC_DATE_TIME_FORMATTER.parse(dateFormat)).withZoneSameInstant(UTC);
+        int separatorIdx = dateFormat.indexOf('-');
+        if (separatorIdx == 0) { // negative year
+            separatorIdx = dateFormat.indexOf('-', 1);
+        }
+        separatorIdx = dateFormat.indexOf('-', separatorIdx + 1) + 3;
```

Comment:
Why '+3' here?

Reply:
if you have a date like `-1000-11-22T10:20:30....`:
check for the first char, if `-` move to idx 1.
Then detect the first `-`, then detect the second `-` and `+3` to move past `22` to the `T` or whitespace.

Reply:
Worth a comment in the code.